superset/common/query_context.py

```diff
@@ -56,7 +56,7 @@ class QueryContext:
     cache_type: ClassVar[str] = "df"
     enforce_numerical_metrics: ClassVar[bool] = True
 
-    datasource: BaseDatasource
+    datasource: SqlaTable
```

Comment:
I think this was discussed in a previous review on some other class. I'm pretty sure we can't assume that a `datasource` is `SqlaTable` here, as `QueryContext` needs to support all supported `BaseDatasource`s.

Reply:
Right. Couldn't remember why I changed this. But whatever it was, it is probably not needed any more.

I remember there were some discussion raised by @john-bodley around deprecating Druid (and maybe other Datasource types) to simplify future maintenance. Was there ever a conclusion from that, John?

Reply:
My impression is that no new features will be added, but we are still not able to fully remove support quite yet. At any rate, as long as we want to at least entertain the thought of supporting non-SQLA sources (e.g. graph databases come to mind), I think we'll need to keep this as `BaseDatasource`.